docs/source/performance/cache-backends.mdx

```diff
@@ -0,0 +1,80 @@
+
+### Example: using `InMemoryLRUCache`
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // Note: this is equivalent to providing `cache: "bounded"`
+  cache: new InMemoryLRUCache(),
+});
+```
+
+### Example: configuring `InMemoryLRUCache`
+
+In this example, we've increased the default size and provided a default TTL. For more information on the options, see the [`lru-cache` documentation`](https://www.npmjs.com/package/lru-cache).
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // ...
+  cache: new InMemoryLRUCache({
+    // ~100MiB
+    maxSize: Math.pow(2, 20) * 100,
+    // 5 minutes
+    ttl: 300000,
+  }),
+});
+```
+
+## Configuring external caching
+
+Apollo no longer maintains any caching backends directly. Instead, we recommend using the [`keyv`](https://www.npmjs.com/package/keyv) package along with the [`KeyvAdapter`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyvAdapter#keyvadapter-class) class provided by the `@apollo/utils.keyvadapter` package. `KeyvAdapter` simply wraps a `Keyv` instance and implements the `KeyValueCache` interface which is required by Apollo Server.
+
+### Example: using `Keyv` with Redis
+
+```ts
+import Keyv from 'keyv';
+import { KeyvAdapter } from '@apollo/utils.keyvadapter';
+
+const server = new ApolloServer({
+  // ...,
+  cache: new KeyvAdapter(new Keyv('redis://localhost:6379')),
+});
+```
+
+## Implementing your own cache backend
+
+If your requirements are more specialized or you prefer to implement your own cache backend, you can implement the `KeyValueCache` interface and pass it to Apollo Server's constructor.
+
+The interface is very simple:
+```ts
+interface KeyValueCache<V> {
+  get(key: string): Promise<V | undefined>;
+  // ttl is specified in seconds
+  set(key: string, value: V, options?: { ttl?: number | null }): Promise<void>;
+  delete(key: string): Promise<boolean | void>;
+}
+```
+
+## Legacy caching implementation
+
+Apollo Server < 3.9.0 used the `apollo-server-caching` package to implement caching. This package is no longer maintained and is not recommended for use. The `KeyValueCache` interface has been moved to the `@apollo/utils.keyvaluecache` package. The `InMemoryLRUCache` class has been moved there as well, however it now uses version 7 of `lru-cache` which accepts different options and no longer allows a cache to be unbounded. The `apollo-server-cache-redis` and `apollo-server-cache-memcached` packages are no longer being published; we recommend using `keyv` instead.
```

Comment:
This is a lot of great info, and I think splitting it up will make it more digestible. 
```suggestion
Before Apollo Server 3.9+, we used the `apollo-server-caching` package to implement caching. The `apollo-server-caching` package is no longer maintained, and we do not recommend using it. The `KeyValueCache` interface has been moved and is now in the `@apollo/utils.keyvaluecache` package. 

The `InMemoryLRUCache` class has also moved to the `@apollo/utils.keyvaluecache` package. The `InMemoryLRUCache` class now uses version 7 of `lru-cache`, accepting different configuration options and no longer allowing a cache to be unbounded.

The `apollo-server-cache-redis` and `apollo-server-cache-memcached` packages are no longer published; we recommend using `keyv` instead.
```